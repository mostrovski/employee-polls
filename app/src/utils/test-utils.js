// https://redux.js.org/usage/writing-tests#setting-up-a-reusable-test-render-function

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { setupStore } from '../app/store';

export const renderWithProviders = (
    component,
    {
        preloadedState = {},
        route = { currentPath: '/', matchPath: '/', wrap: false },
        store = setupStore(preloadedState),
        ...renderOptions
    } = {}
) => {
    const Wrapper = ({ children }) => {
        return (
            <Provider store={store}>
                <MemoryRouter initialEntries={[`${route.currentPath}`]}>
                    {children}
                </MemoryRouter>
            </Provider>
        );
    };

    if (route.wrap) {
        return {
            store,
            ...render(
                <Routes>
                    <Route path={route.matchPath} element={component} />
                </Routes>,
                { wrapper: Wrapper, ...renderOptions }
            ),
        };
    }

    return {
        store,
        ...render(component, { wrapper: Wrapper, ...renderOptions }),
    };
};
