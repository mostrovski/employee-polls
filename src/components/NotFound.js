import { NavLink } from 'react-router-dom';

export default function NotFound() {
    return (
        <div className="flex min-h-full items-center justify-center py-12 px-4 sm:px-6 lg:px-8">
            <div className="w-full max-w-md space-y-8">
                <div>
                    <img
                        className="mx-auto h-12 w-auto rounded"
                        src="/logo512.png"
                        alt="Your Company"
                    />
                    <h2 className="mt-6 text-center text-3xl font-bold tracking-tight text-gray-900">
                        Failed to find anything...
                    </h2>
                </div>

                <div className="text-center">
                    <NavLink
                        to="/"
                        className="font-medium text-violet-600 hover:text-violet-500"
                    >
                        Sign in to your account
                    </NavLink>
                </div>
            </div>
        </div>
    );
}
