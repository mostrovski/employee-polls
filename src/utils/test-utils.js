// https://redux.js.org/usage/writing-tests#setting-up-a-reusable-test-render-function

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { setupStore } from '../app/store';

export const renderWithProviders = (
    component,
    {
        preloadedState = {},
        store = setupStore(preloadedState),
        ...renderOptions
    } = {}
) => {
    const Wrapper = ({ children }) => {
        return (
            <Provider store={store}>
                <MemoryRouter>{children}</MemoryRouter>
            </Provider>
        );
    };

    return {
        store,
        ...render(component, { wrapper: Wrapper, ...renderOptions }),
    };
};
