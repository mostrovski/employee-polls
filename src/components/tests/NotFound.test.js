import { renderWithProviders } from '../../utils/test-utils';
import NotFound from '../NotFound';

it('renders correctly', () => {
    const view = renderWithProviders(<NotFound />);

    expect(view).toMatchSnapshot();
});
