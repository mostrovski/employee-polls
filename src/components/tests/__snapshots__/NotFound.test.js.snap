// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex min-h-full items-center justify-center py-12 px-4 sm:px-6 lg:px-8"
      >
        <div
          class="w-full max-w-md space-y-8"
        >
          <div>
            <img
              alt="Employee Polls"
              class="mx-auto h-14 w-auto rounded"
              src="/logo512.png"
            />
            <h2
              class="mt-6 text-center text-3xl font-bold tracking-tight text-gray-900"
            >
              Failed to find anything...
            </h2>
          </div>
          <div
            class="text-center"
          >
            <a
              class="link-primary"
              href="/"
            >
              Sign in to your account
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex min-h-full items-center justify-center py-12 px-4 sm:px-6 lg:px-8"
    >
      <div
        class="w-full max-w-md space-y-8"
      >
        <div>
          <img
            alt="Employee Polls"
            class="mx-auto h-14 w-auto rounded"
            src="/logo512.png"
          />
          <h2
            class="mt-6 text-center text-3xl font-bold tracking-tight text-gray-900"
          >
            Failed to find anything...
          </h2>
        </div>
        <div
          class="text-center"
        >
          <a
            class="link-primary"
            href="/"
          >
            Sign in to your account
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
